test/gscraper_web/views/shared_view_test.exs

```diff
@@ -0,0 +1,45 @@
+defmodule GscraperWeb.SharedViewTest do
+  use GscraperWeb.ConnCase, async: true
+
+  alias GscraperWeb.SharedView
+
+  describe "current_user/1" do
+    test "returns the current user object from the connection given it exists", %{conn: conn} do
+      conn = Plug.Conn.put_private(conn, :guardian_default_resource, :mock_user)
+
+      assert SharedView.current_user(conn) == :mock_user
+    end
+
+    test "returns nil given no current user object in the connection", %{conn: conn} do
+      assert SharedView.current_user(conn) == nil
+    end
+  end
+
+  describe "current_user_username/1" do
+    test "returns the current user's username from conn given the current user object exists in the conn",
```

Comment:
Maybe this description can be simplified too

Reply:
Also fixed in dcdd8cf.